### ISSUE TITLE
Picture encoding/feature extraction

### ISSUE THREAD

**Issue by raphaeli** [vlfeat/matconvnet#64]
Hi,
I was wondering if you could show how to encode a picture using the pretrained models by bypassing the output layer. Would it be as simple as putting
net.layers{end}.type = 'softmax' to some layer? For example in vgg verydeep from which layer should we draw features?

Thank you!
%%%%%%%%%%%%%%
function cnn_imagenet_minimal()
% CNN_IMAGENET_MINIMAL   Minimalistic demonstration of how to run an ImageNet CNN model

% setup toolbox
run(fullfile(fileparts(mfilename('fullpath')), ...
  '..', 'matlab', 'vl_setupnn.m')) ;

% download a pre-trained CNN from the web
if ~exist('iimagenet-vgg-verydeep-19', 'file')
  urlwrite('http://www.vlfeat.org/matconvnet/models/imagenet-vgg-verydeep-19.mat', ...
    'imagenet-vgg-verydeep-19.mat') ;
end

net = load('imagenet-vgg-verydeep-19.mat') ;
% obtain and preprocess an image
im = imread('peppers.png') ;
im_ = single(im) ; % note: 255 range
im_ = imresize(im_, net.normalization.imageSize(1:2)) ;
im_ = im_ - net.normalization.averageImage ;

% run the CNN
res = vl_simplenn(net, im_) ;

% show the classification result
scores = squeeze(gather(res(end).x)) ;
[bestScore, best] = max(scores) ;
figure(1) ; clf ; imagesc(im) ;
title(sprintf('%s (%d), score %.3f',...
   net.classes.description{best}, best, bestScore)) ;

**Comment by lenck** [comment]
Hi, you can simply remove completely the output layer (softmax), e.g. adjusting the net:
```
net.layers = net.layer(1:end-1);
```
before you cal the simplenn. Then you would have at the end the the output of the FC8. In a simlar manner, if you just need e.g. the convx features, you can remove even more layers to speed-up the computation.

**Comment by arvieFrydenlund** [comment]
Thanks.  That is very helpful.

**Comment by sunilrao43** [comment]
@vedaldi  how can run localization task vgg pretrained model 
 thanks in advance

**Comment by lenck** [comment]
@sunilrao43 sorry, but I think your question is a bit off-topic here. Can you post it on the discussion group?
I guess this issue can be closed...
